src/ctrlc.rs

```diff
@@ -0,0 +1,73 @@
+use console::Term;
+use lazy_static::lazy_static;
+use signal_hook::{
+    consts::SIGINT,
+    iterator::{Handle, Signals},
+};
+use std::{
+    io::Error,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Mutex, RwLock,
+    },
+    thread,
+};
+
+static INIT: AtomicBool = AtomicBool::new(false);
+static INIT_LOCK: Mutex<()> = Mutex::new(());
+
+lazy_static! {
+    static ref HANDLE: RwLock<Option<Handle>> = RwLock::new(None);
+}
+
+pub fn show_cursor_after_ctrlc(term: &Term) -> Result<Option<Handle>, Error> {
+    let t = term.clone();
+    set_ctrlc_handler(move || {
+        println!("Received Ctrl-C");
+        t.show_cursor().unwrap();
+    })
+}
+
+pub fn close_ctrlc_handle(handle: Option<Handle>) {
+    if let Some(handle) = handle {
+        handle.close();
+    }
+}
+
+pub fn set_ctrlc_handler<F>(mut handler: F) -> Result<Option<Handle>, Error>
+where
+    F: FnMut() + 'static + Send,
+{
+    let _guard = INIT_LOCK.lock();
+    if INIT.load(Ordering::Relaxed) {
+        let handle_guard = HANDLE.read().unwrap();
+        return Ok(handle_guard.clone());
+    }
+    INIT.store(true, Ordering::Relaxed);
```

Comment:
ok I tested this and it does seem to work as you described.

As far as the multiple prompts things, I handle this with a mutex in mise: https://github.com/jdx/mise/blob/0f9dae377eaa8a527c099310944e75bac60bc3a3/src/ui/prompt.rs#L13

but we should probably put that in here since it never makes sense to show 2 prompts at the same time